state-chain/chains/src/sol/sol_tx_core.rs

```diff
@@ -1090,30 +1066,18 @@ mod tests {
 				&Pubkey::from_str(NONCE_ACCOUNTS[0]).unwrap(),
 				&agg_key_pubkey,
 			),
-			ProgramInstruction::get_instruction(
-				&VaultProgram::FetchTokens {
-					seed: vec![19u8, 2u8, 11u8],
-					bump: 254,
-					amount: 100000000,
-					decimals: 6,
-				},
-				Pubkey::from_str(VAULT_PROGRAM).unwrap(),
-				vec![
-					AccountMeta::new_readonly(
-						Pubkey::from_str(VAULT_PROGRAM_DATA_ACCOUNT).unwrap(),
-						false,
-					),
-					AccountMeta::new_readonly(agg_key_pubkey, true),
-					AccountMeta::new_readonly(deposit_channel, false),
-					AccountMeta::new(deposit_channel_ata, false),
-					AccountMeta::new(
-						Pubkey::from_str(TOKEN_VAULT_ASSOCIATED_TOKEN_ACCOUNT).unwrap(),
-						false,
-					),
-					AccountMeta::new_readonly(Pubkey::from_str(MINT_PUB_KEY).unwrap(), false),
-					AccountMeta::new_readonly(Pubkey::from_str(TOKEN_PROGRAM_ID).unwrap(), false),
-					AccountMeta::new_readonly(Pubkey::from_str(SYSTEM_PROGRAM_ID).unwrap(), false),
```

Comment:
FYI I think the difference might come from here. The old code had the system_program_id, the idl definitions don't contain this. Everything else is the same.

Reply:
I think this is related to a gotcha that Solana has, where the System program is required for some calls but it's not required anywhere, so you need to know that it's required and pass it, otherwise it will fail. The new Anchor IDL probably messed that one up.
In any case, fetch tokens is exactly the one that needs several changes so I'll have to review it. Thanks for the info.